lib/3scale/backend/async_redis/endpoint_helpers.rb

```diff
@@ -14,20 +15,40 @@ class << self
 
           # @param host [String]
           # @param port [Integer]
+          # @param path [String]
+          # @param ssl [Boolean]
           # @param ssl_params [Hash]
           # @return [Async::IO::Endpoint::Generic]
-          def prepare_endpoint(host, port, ssl = false, ssl_params = nil)
+          def prepare_endpoint(**kwargs)
+            return prepare_tcp_endpoint(**kwargs) if host_present?(kwargs[:host])
+
+            prepare_unix_endpoint(**kwargs)
+          end
+
+          private
+
+          def prepare_tcp_endpoint(host: nil, port: nil, ssl: false, ssl_params: nil)
             tcp_endpoint = Async::IO::Endpoint.tcp(host, port)
 
-            if ssl
-              ssl_context = OpenSSL::SSL::SSLContext.new
-              ssl_context.set_params(format_ssl_params(ssl_params)) if ssl_params
-              return Async::IO::SSLEndpoint.new(tcp_endpoint, ssl_context: ssl_context)
-            end
+            return prepare_ssl_endpoint(endpoint: tcp_endpoint, ssl_params: ssl_params) if ssl
 
             tcp_endpoint
           end
 
+          def prepare_unix_endpoint(path: '', ssl: false, ssl_params: nil)
+            unix_endpoint = Async::IO::Endpoint.unix(path, Socket::PF_UNIX)
+
+            return prepare_ssl_endpoint(endpoint: unix_endpoint, ssl_params: ssl_params) if ssl
+
+            unix_endpoint
```

Comment:
A little bit effort to understand. Easy to miss this `if ssl` at the end of the long line. I'd suggest one of those.
```suggestion
            unix_endpoint = Async::IO::Endpoint.unix(path, Socket::PF_UNIX)
            return unix_endpoint unless ssl

            prepare_ssl_endpoint(endpoint: unix_endpoint, ssl_params: ssl_params)
```

or

```suggestion
            unix_endpoint = Async::IO::Endpoint.unix(path, Socket::PF_UNIX)

            ssl ? prepare_ssl_endpoint(endpoint: unix_endpoint, ssl_params: ssl_params) : unix_endpoint
```